src/main/java/org/candlepin/testext/hostedtest/HostedTestCloudRegistrationAdapter.java

```diff
@@ -134,10 +138,9 @@ public CloudAuthenticationResult resolveCloudRegistrationDataV2(CloudRegistratio
         }
 
         String offerType = this.datastore.getOfferTypeForOfferId(offerId);
-        if (OFFERING_TYPE_1P.equals(offerType)) {
-            throw new CloudRegistrationNotSupportedForOfferingException(
-                "cloud registration v2 is not supported for 1P offerings");
-        }
+        boolean isRegistrationOnly = offerType == null ?
+            false :
+            REGISTRATION_ONLY_OFFER_TYPES.contains(offerType);
```

Comment:
For the same reason I mentioned we still need a `CloudRegistrationNotSupportedForOfferingException`, we need some updated logic here. Something like:
```
if (REGISTRATION_ONLY_OFFER_TYPES.contains(offerType)) {
    if (this.datastore.getOwnerKeyForCloudAccountId(accountId) != null) {
        // we have a mapping, even though its 1P
        isRegistrationOnly = true;
    }
    else {
         // no mapping
         throw new CloudRegistrationNotSupportedForOfferingException(
             "cloud registration v2 is not supported for 1P offerings");
    }
}

```

Reply:
Also, this means we want an additional spec test. `shouldReceiveStandardTokenWithRegistrationOnlyOfferTypesForV2Auth` tests for the registration-only scenario because `associateOwnerToCloudAccount` was used (we have a mapping in the cloud access db), so we need another one without `associateOwnerToCloudAccount`  (we have no mapping), where `NotImplementedException` is expected

Reply:
I addressed this in the other [comment](https://github.com/candlepin/candlepin/pull/4565#discussion_r1494937035), but I think that in this case we should be returning a `NotAuthorizedException` and not `NotImplementedException`, because the problem is with the cloud account and org association and not the cloud offering.

Reply:
Based on the other comment, I updated to use this logic.